enginetest/queries/ansi_quotes_queries.go

```diff
@@ -0,0 +1,316 @@
+			},
+			{
+				Query:          `select """""foo""""";`,
+				ExpectedErrStr: `column "\"\"foo\"\"" could not be found in any table in scope`,
+			},
+			{
+				Query:          "select ```foo```;",
+				ExpectedErrStr: "column \"`foo`\" could not be found in any table in scope",
+			},
+			{
+				// Disable ANSI_QUOTES and make sure we can still run queries
+				Query:    `SET @@sql_mode='NO_ENGINE_SUBSTITUTION,ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES';`,
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    `select "data" from auctions order by "ai" desc;`,
+				Expected: []sql.Row{{"data"}},
+			},
+			{
+				Query:    `show tables;`,
+				Expected: []sql.Row{{"auctions"}, {"myview"}},
+			},
+		},
+	},
+	{
+		Name: "ANSI_QUOTES: ANSI mode includes ANSI_QUOTES",
+		SetUpScript: []string{
+			`SET @@sql_mode='ANSI';`,
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				// Assert that we can create a table using ANSI style quotes
+				Query:    `create table "t" ("pk" int primary key, "data" varchar(100));`,
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    `insert into t ("pk", "data") values (1, 'one');`,
+				Expected: []sql.Row{{types.NewOkResult(1)}},
+			},
+			{
+				Query:    `select "pk", "data" from "t" order by "pk" asc;`,
+				Expected: []sql.Row{{1, "one"}},
+			},
+		},
+	},
+	{
+		Name: "ANSI_QUOTES: views",
+		SetUpScript: []string{
+			`SET @@sql_mode='ANSI_QUOTES,NO_ENGINE_SUBSTITUTION,ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES';`,
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				// https://github.com/dolthub/dolt/issues/6305
+				Query:    `CREATE TABLE public_keys (item INTEGER, type CHAR(4), hash INTEGER, "count" INTEGER, "public" VARCHAR(8000))`,
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    `create view view1 as select public_keys."public", public_keys."count" from public_keys;`,
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    `show tables;`,
+				Expected: []sql.Row{{"myview"}, {"public_keys"}, {"view1"}},
+			},
+			{
+				Query:    `show create table view1;`,
+				Expected: []sql.Row{{"view1", "CREATE VIEW `view1` AS select public_keys.\"public\", public_keys.\"count\" from public_keys", "utf8mb4", "utf8mb4_0900_bin"}},
+			},
+			{
+				// Assert that we can load and parse views for information_schema when ANSI_QUOTES mode is enabled
+				Query:    `select table_name, view_definition from information_schema.views where table_name='view1';`,
+				Expected: []sql.Row{{"view1", `select public_keys."public", public_keys."count" from public_keys`}},
+			},
+			{
+				// Disable ANSI_QUOTES mode
+				Query:    `SET @@sql_mode='NO_ENGINE_SUBSTITUTION,ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES';`,
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    `show create table view1;`,
+				Expected: []sql.Row{{"view1", "CREATE VIEW `view1` AS select public_keys.\"public\", public_keys.\"count\" from public_keys", "utf8mb4", "utf8mb4_0900_bin"}},
```

Comment:
Good question... this is slightly different than MySQL's behavior – MySQL will normalize the view definition for the SQL_MODE the session is in. So.... if `ANSI_QUOTES` is enabled, then double quotes will be shown around identifiers, otherwise backtick quotes will be shown in the returned definition.  

I'll add some skipped tests and document this as a gap in the current support. I'd vote to leave this as a follow-up, but let me know if you think it's higher priority than that.